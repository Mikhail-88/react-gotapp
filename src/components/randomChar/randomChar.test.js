import React from 'react';
import RandomChar from './randomChar';
import { shallow } from 'enzyme';

describe('Testing <RandomChar/>', () => {
  const charComponent = shallow(<RandomChar />);

  describe('Testing snap and state', () => {
    it('RandomChar have rendered correctly', () => {
      expect(charComponent).toMatchSnapshot();
    });

    it('RandomChar state "character" is empty object', () => {
      expect(charComponent.state().character).toBeObject();
    });

    it('RandomChar state "isLoading" is true', () => {
      expect(charComponent.state().isLoading).toBeTruthy();
    });

    it('RandomChar state "error" is false', () => {
      expect(charComponent.state().error).toBeFalsy();
    });
  });

  describe('Handlers tests', () => {
    it('testing onCharacterLoaded', () => {
      charComponent.instance().onCharacterLoaded();
      expect(charComponent.state().isLoading).toBeFalsy();
    });

    it('testing onError', () => {
      charComponent.instance().onError();
      expect(charComponent.state().error).toBeTruthy();
      expect(charComponent.state().isLoading).toBeFalsy();
    });

    it('testing updateCharacter', () => {
      charComponent.instance().updateCharacter();
      expect(charComponent.state().isLoading).toBeFalsy();
    });
  });
});