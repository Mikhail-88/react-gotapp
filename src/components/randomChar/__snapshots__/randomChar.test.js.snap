// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RandomChar/> Testing snap and state RandomChar have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RandomChar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Random Character
        </h4>,
        <Spinner />,
        <Button
          color="info"
          onClick={[Function]}
          outline={true}
          tag="button"
        >
          Next character
        </Button>,
      ],
      "className": "random-block rounded",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Random Character",
        },
        "ref": null,
        "rendered": "Random Character",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Next character",
          "color": "info",
          "onClick": [Function],
          "outline": true,
          "tag": "button",
        },
        "ref": null,
        "rendered": "Next character",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Random Character
          </h4>,
          <Spinner />,
          <Button
            color="info"
            onClick={[Function]}
            outline={true}
            tag="button"
          >
            Next character
          </Button>,
        ],
        "className": "random-block rounded",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Random Character",
          },
          "ref": null,
          "rendered": "Random Character",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Next character",
            "color": "info",
            "onClick": [Function],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": "Next character",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
